sim.c

```diff
@@ -463,11 +463,13 @@ BEGIN_OPERATOR(delay)
   PORT(0, -1, IN | PARAM);
   PORT(0, 1, IN);
   PORT(1, 0, OUT);
+  Usz mod_num = index_of(PEEK(0, 1));
   Usz rate = index_of(PEEK(0, -1));
-  Usz offset = index_of(PEEK(0, 1));
```

Comment:
Prefer to read from the lower addresses first, then work up towards higher addresses -- `PEEK(0, -1)` will be at a lower address than `PEEK(0, 1)`. This is generally easier on the memory prefetcher.

Reply:
i just copied the code from clock, so we have to invert the PEEKs also there

Reply:
Correct, looks like they got messed up at some point.